bentoml/metrics.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import typing as t
+import logging
+from typing import TYPE_CHECKING
+
+from simple_di import inject
+from simple_di import Provide
+
+logger = logging.getLogger(__name__)
+
+from ._internal.configuration.containers import BentoMLContainer
+
+if TYPE_CHECKING:
+    from bentoml._internal.server.metrics.prometheus import PrometheusClient
+
+
+class BaseMetric(object):
```

Comment:
Make this class a abc class
```suggestion

from abc import ABC, abstractmethod
class MetricMeta(ABC):
```

Reply:
I made BaseMetric a base class with abstractmethod. I almost updated it to make it 'protected'.